docs/features.md

```diff
@@ -40,13 +40,24 @@ Example:
     "caption": "Thread First",
     "command": "lsp_execute",
     "args": { 
-    	"command_name": "thread-first",
-    	"command_args":["file:///tmp/foo.clj", 0, 0] 
+      "command_name": "thread-first",
+      "command_args": ["${file_uri}", 0, 0]
     }
   }
 ]
 ```
-Note: `command_args` is optional depending on the `workspace/executeCommand` that are supported by the LSP server. 
+Note: `command_args` is optional depending on the `workspace/executeCommand` that are supported by the LSP server.
+The following variables will be expanded, but only if they are top-level array items and not within nested arrays or objects:
+
+| Variable | Type | Description |
+| -------- | ---- | ----------- |
+| `"$file_uri"` or `"${file_uri}"` | string | File URI of the active view |
+| `"$selection"` or `"${selection}"` | string | Content of the (rearmost) selection |
```

Comment:
Rearmost: usually we reason from the first selection instead of the last selection (e.g. completions take the first cursor position as the point to request completions for). Why do we have to take the last selection/cursor for commands?

Reply:
From a usage perspective, I would argue that the rearmost is the better choice. When I use multiple cursors/selections, I usually add them from left to right and from top to bottom of a file. Then, the rearmost is the last added or "active" one, and when I suddenly see that I want to run the workspace/executeCommand for that selection, I don't have to reset all other selections first to do it.

However if we use the first selection in all other cases, I will change it here to the first as well for consistency. I think to always use the last cursor/selection would be the better way though.

Reply:
The AST is in most cases wrong when using the last cursor position for completions. For commands this probably doesn't matter as the text bufer is most likely in a good state (no errors). For consistency's sake I still think we should take the first region/cursor of the selection to avoid surprises for users.

Reply:
Yes I agree, changed it to the first.